apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsAnyOfControl.tsx

```diff
@@ -35,25 +36,36 @@ export function JsonFormsAnyOfControl({
     uischemas,
   )
 
-  const variants = anyOfRenderInfos.map((anyOfRenderInfo) => ({
-    label: anyOfRenderInfo.label,
-    value: anyOfRenderInfo.label,
-  }))
+  const options = anyOfRenderInfos.map((anyOfRenderInfo) => {
+    const option = String(anyOfRenderInfo.schema.const || anyOfRenderInfo.label)
 
-  const [variant, setVariant] = useState(anyOfRenderInfos[0]?.label || "")
+    return {
+      label: option.charAt(0).toUpperCase() + option.slice(1),
+      value: option,
+    }
+  })
+
+  const [variant, setVariant] = useState("")
+
+  const onChange = (value: string) => {
+    setVariant(value)
+    handleChange(path, value)
+  }
 
   return (
-    <Box py={2}>
-      <FormControl isRequired>
-        <FormLabel>Variant</FormLabel>
-        <SingleSelect
-          value={variant}
-          name={label}
-          items={variants}
-          isClearable={false}
-          onChange={setVariant}
-        />
-      </FormControl>
+    <>
+      <Box py={2}>
```

Comment:
nit: prefer explicit declaration for consistency.

```suggestion
      <Box py="<xx>rem"> 
```

in the future might have specific spacing tokens haha

Reply:
Ah I wasn't sure if we should use rem or numbers but agree seems like rem is a lot easier to reason. Fixed in 993b9c92c9f4d1c3a6a5405ef52321df66759a0e!